code/script.js

```diff
@@ -231,95 +233,138 @@ const setSecret = () => {
+        `No, the person doesn't wear ${value}! Removing all people that wear ${value}.`
       )
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
     }
+
   } else if (category === 'other') {
-    // Similar to the one above
-  } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(
+        `Yes, the person has ${value}! Keeping all people with ${value}.`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `No, the person doesn't have ${value}! Removing all people with ${value}.`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+    }  
+  } else if (category === 'hair'){
+    if (keep) {
+      alert(
+        `Yes, the person has ${value} ${category}! Keeping all people that have ${value} ${category}.`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+    } else {
+      alert(
+        `No, the person doesn't have ${value} ${category}! Removing all people that have ${value} ${category}.`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
     }
-  }
-
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
+  } else if (category === 'eyes') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${value} ${category}! Keeping all people that have ${value} ${category}.`
+      )
       charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
+    } else {
+      alert(
+        `No, the person doesn't have ${value} ${category}! Removing all people that have ${value} ${category}.`
+      )
       charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+    }
+  }
 
-  // Invoke a function to redraw the board with the remaining people.
+//Redraws the board with the people left to choose from.
+  generateBoard()
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  choosePerson = confirm (`Guess on ${personToConfirm}?`)
+  if (choosePerson) {
```

Comment:
Here you have started a conditional but it doesn't lead anywhere? Or what have I missed?